BeagleBone_black/src/motor_control/controller.cpp

```diff
@@ -198,117 +226,214 @@ void Controller::configure()
-SDOMessage.data[0]   = kWRITE_2_BYTES;
-SDOMessage.data[1]   = 0x40;
-SDOMessage.data[2]   = 0x60;
-SDOMessage.data[3]   = 0x00;
-SDOMessage.data[4]   = 0x06;
-SDOMessage.data[5]   = 0x00;
-SDOMessage.data[6]   = 0x00;
-SDOMessage.data[7]   = 0x00;
-
-sendSdoCan(SDOMessage);
-log_.DBG1("MOTOR", "Controller %d: Shutdown command sent", node_id_);
-
-while (state_ == kSwitchOnDisabled);
-this->checkStatus();
-
-// Send switch on message to transition from state 2 (Ready to switch on)
-// to state 3 (Switched on)
-SDOMessage.data[0]   = kWRITE_2_BYTES;
-SDOMessage.data[1]   = 0x40;
-SDOMessage.data[2]   = 0x60;
-SDOMessage.data[3]   = 0x00;
-SDOMessage.data[4]   = 0x07;
-SDOMessage.data[5]   = 0x00;
-SDOMessage.data[6]   = 0x00;
-SDOMessage.data[7]   = 0x00;
-
-sendSdoCan(SDOMessage);
-log_.DBG1("MOTOR", "Controller %d: Switch on command sent", node_id_);
-
-while (state_ == kReadyToSwitchOn);
-this->checkStatus();
-
-// Send enter operational message to transition from state 3 (Switched on)
-// to state 4 (Operation enabled)
-SDOMessage.data[0]   = kWRITE_2_BYTES;
-SDOMessage.data[1]   = 0x40;
-SDOMessage.data[2]   = 0x60;
-SDOMessage.data[3]   = 0x00;
-SDOMessage.data[4]   = 0x0F;
-SDOMessage.data[5]   = 0x00;
-SDOMessage.data[6]   = 0x00;
-SDOMessage.data[7]   = 0x00;
-
-sendSdoCan(SDOMessage);
-log_.DBG1("MOTOR", "Controller %d: Enabling drive function", node_id_);
-
-while (state_ == kSwitchedOn);
-this->checkStatus();
-
-// Check warning status
-SDOMessage.data[0]   = kREAD_OBJECT;
-SDOMessage.data[1]   = 0x27;
-SDOMessage.data[2]   = 0x20;
-SDOMessage.data[3]   = 0x00;
-SDOMessage.data[4]   = 0x00;
-SDOMessage.data[5]   = 0x00;
-SDOMessage.data[6]   = 0x00;
-SDOMessage.data[7]   = 0x00;
-
-sendSdoCan(SDOMessage);
-log_.INFO("MOTOR", "Controller %d: Checking for warnings", node_id_);
-
-// Check error status
-SDOMessage.data[0]   = kREAD_OBJECT;
-SDOMessage.data[1]   = 0x3F;
-SDOMessage.data[2]   = 0x60;
-SDOMessage.data[3]   = 0x00;
-SDOMessage.data[4]   = 0x00;
-SDOMessage.data[5]   = 0x00;
-SDOMessage.data[6]   = 0x00;
-SDOMessage.data[7]   = 0x00;
-
-sendSdoCan(SDOMessage);
-log_.INFO("MOTOR", "Controller %d: Checking for errors", node_id_);
+  uint8_t state_count;
+
+  // Send NMT Operational message to tranition from state 0 (Not ready to switch on)
+  // to state 1 (Switch on disabled)
+  NMTMessage.data[0]   = kNMT_OPERATIONAL;
+  NMTMessage.data[1]   = node_id_;
+
+  log_.INFO("MOTOR", "Controller %d: Sending NMT Operational command", node_id_);
+  can_.send(NMTMessage);
+
+  // Enable velocity mode
+  SDOMessage.data[0]   = kWRITE_1_BYTE;
+  SDOMessage.data[1]   = 0x60;
+  SDOMessage.data[2]   = 0x60;
+  SDOMessage.data[3]   = 0x00;
+  SDOMessage.data[4]   = 0x09;
+  SDOMessage.data[5]   = 0x00;
+  SDOMessage.data[6]   = 0x00;
+  SDOMessage.data[7]   = 0x00;
+
+  log_.DBG1("MOTOR", "Controller %d: Enabling velocity mode", node_id_);
+  sendSDO(SDOMessage);
+
+  // Set target velocity to 0;
+  this->sendTargetVelocity(0);
```

Comment:
`this->` shouldn't be needed here. Consider removing or adding it everywhere consistently.